pcg/src/main/kotlin/nl/perfana/config/generator/PerfanaConfigGeneratorController.kt

```diff
@@ -46,11 +35,29 @@ class PerfanaConfigGeneratorController(val storage: FileStorage) {
             workDir,
             "source /root/.bashrc && /files/generate-config.sh $zipFileName $generatedFilesDir"
         )
+        val perfana_client = readPerfanaClientValue(workDir)
+
+        // create the projectId
+        val projectId = "perfana-starter-${perfana_client}-${timeStamp}"
 
         // now zip generated files
         val zipFile = storage.zip(workDir.resolve(generatedFilesDir), workDir, projectId)
         // keep in mapping for future download
         zipFiles[projectId] = zipFile
+
+        return projectId
+    }
+
+    private fun readPerfanaClientValue(workDir: Path): String {
+        // get the value of the PERFANA_CLIENT variable from setup.sh
+        val shellCommand = listOf("/bin/bash", "-c", "source ${workDir}/unzip/setup.sh && echo \$PERFANA_CLIENT")
```

Comment:
seems a bit "heavy" to get the client name, but does the job

Reply:
I know. But it is the only simple way to get the value of the PERFANA_CLIENT env var. The only other place where it is available, is when executing the transform.kts.